ext/TrixiNLsolveExt.jl

```diff
@@ -0,0 +1,135 @@
+# Package extension for adding NLsolve-based features to Trixi.jl
+module TrixiNLsolveExt
+
+# Required for finding coefficients in Butcher tableau in the third order of 
+# P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using NLsolve: nlsolve
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..NLsolve: nlsolve
+end
+
+# We use a random initialization of the nonlinear solver.
+# To make the tests reproducible, we need to seed the RNG
+using StableRNGs: StableRNG, rand
+
+# Use functions that are to be extended and additional symbols that are not exported
+using Trixi: Trixi, compute_c_coeffs, @muladd
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# Compute residuals for nonlinear equations to match a stability polynomial with given coefficients,
+# in order to find A-matrix in the Butcher-Tableau
+function PairedExplicitRK3_butcher_tableau_objective_function(a_unknown, num_stages,
+                                                              num_stage_evals,
+                                                              monomial_coeffs, c_eq,
+                                                              cS2)
```

Comment:
```suggestion
function PairedExplicitRK3_butcher_tableau_objective_function!(c_eq, a_unknown, num_stages,
                                                              num_stage_evals,
                                                              monomial_coeffs,
                                                              cS2)
```
I think, `c_eq` should come first as this is the argument that gets mutated by the function (thus also the `!`).